django/conf/__init__.py

```diff
@@ -181,7 +182,13 @@ def __init__(self, settings_module):
         # store the settings module in case someone later cares
         self.SETTINGS_MODULE = settings_module
 
-        mod = importlib.import_module(self.SETTINGS_MODULE)
+        try:
+            mod = importlib.import_module(self.SETTINGS_MODULE)
+        except ImportError as exc:
+            # If the settings module cannot be imported, treat it as a configuration error.
+            if not self.was_settings_module_found():
```

Comment:
This seems unnecessarily complicated? Maybe I'm missing something, but what about this?
```suggestion
        except ModuleNotFoundError as exc:
            # If the settings module cannot be imported, treat it as a configuration error.
            if module.name == self.SETTINGS_MODULE:
```

Reply:
Yes, but with `exc.name` 😉

Reply:
Ha right thanks. Fixed.